data-migrator/core/src/main/java/io/camunda/migration/data/HistoryMigrator.java

```diff
@@ -332,72 +474,181 @@ protected void migrateDecisionInstance(HistoricDecisionInstance c7DecisionInstan
+        // Validate process instance is migrated and fetch key
+        if (shouldSkipDecisionInstanceDueToMissingProcessInstance(c7DecisionInstance, context)) {
+          return;
+        }
+        ProcessInstanceEntity processInstance = findProcessInstanceByC7Id(c7DecisionInstance.getProcessInstanceId());
+        if (processInstance != null && processInstance.processInstanceKey() != null) {
+          decisionInstanceDbModelBuilder.processInstanceKey(processInstance.processInstanceKey());
+        }
 
-      try {
+        // Handle root decision instance (parent) if present
+        String c7RootDecisionInstanceId = c7DecisionInstance.getRootDecisionInstanceId();
+        Long parentDecisionDefinitionKey = null;
+        if (c7RootDecisionInstanceId != null) {
+          if (shouldSkipDecisionInstanceDueToMissingParent(c7RootDecisionInstanceId, c7DecisionInstance, context)) {
+            return;
+          }
+          DecisionInstanceEntity parentDecision = findDecisionInstance(c7RootDecisionInstanceId);
+          if (parentDecision != null && parentDecision.decisionDefinitionKey() != null) {
+            parentDecisionDefinitionKey = parentDecision.decisionDefinitionKey();
+            decisionInstanceDbModelBuilder.rootDecisionDefinitionKey(parentDecisionDefinitionKey);
+          }
+        }
+
+        // Validate flow node instance is migrated and fetch entity
+        if (shouldSkipDecisionInstanceDueToMissingFlowNodeInstanceInstance(c7DecisionInstance, context)) {
+          return;
+        }
+        FlowNodeInstanceDbModel flowNode = findFlowNodeInstance(c7DecisionInstance.getActivityInstanceId());
+        if (flowNode != null) {
+          if (flowNode.flowNodeInstanceKey() != null) {
+            decisionInstanceDbModelBuilder.flowNodeInstanceKey(flowNode.flowNodeInstanceKey());
+          }
+          if (flowNode.flowNodeId() != null) {
+            decisionInstanceDbModelBuilder.flowNodeId(flowNode.flowNodeId());
+          }
+        }
+
+        // Generate decision instance key and finalize model
         Long decisionInstanceKey = getNextKey();
+        String decisionInstanceId = String.format("%s-%d", decisionInstanceKey, 1);
         DmnModelInstance dmnModelInstance = c7Client.getDmnModelInstance(c7DecisionInstance.getDecisionDefinitionId());
-        DecisionInstanceDbModel dbModel = decisionInstanceConverter.apply(c7DecisionInstance,
-            String.format("%s-%d", decisionInstanceKey, 1), decisionDefinition.decisionDefinitionKey(),
-            processDefinitionKey, decisionDefinition.decisionRequirementsKey(), processInstanceKey,
-            parentDecisionDefinitionKey, flowNode.flowNodeInstanceKey(), flowNode.flowNodeId(),dmnModelInstance);
+
+        decisionInstanceDbModelBuilder.decisionInstanceId(decisionInstanceId)
+            .decisionInstanceKey(Long.parseLong(decisionInstanceId.split("-")[0]))
```

Comment:
Since this is now here and no longer in the converter class I guess we dont need to do the split logic anymore and can just directly use `decisionInstanceKey`